target/linux/ar71xx/image/senao.mk

```diff
@@ -0,0 +1,30 @@
+define Build/senao-factory-image
+	mkdir -p $(KDIR_TMP)/$(1)
+	touch $(KDIR_TMP)/$(1)/FWINFO-OpenWrt-$(REVISION)-$(1)
+	# $(CP) ./$(1)/* $(KDIR_TMP)/$(1)
+	dd if=$(KDIR)/$(1)-kernel.bin \
+		of=$(KDIR_TMP)/$(1)/openwrt-senao-$(1)-uImage-lzma.bin bs=64k conv=sync
+	dd if=$(KDIR)/root.squashfs \
+		of=$(KDIR_TMP)/$(1)/openwrt-senao-$(1)-root.squashfs bs=64k conv=sync
+	( \
+		cd $(KDIR_TMP)/$(1);  \
+		$(TAR) -cz -f $(call factoryname,squashfs,$(1)) * \
+	)
+	rm -rf $(KDIR_TMP)/$(1)
+endef
+
+define Device/ens202ext
```

Comment:
That wouldn't work as above; the factory.img _has_ to have two separate files with that exact naming scheme, or else the vendor firmware will not accept it for upgrading to even get into LEDE. Now, regarding sysupgrade I don't know, but the factory image (which my understanding says is the way from vendor->lede, correct me if I'm wrong) has to come out the same as my current recipe has (that is, a separate uImage-lzma.bin and root.squashfs with the exact naming scheme).